src/render/batch.rs

```diff
@@ -160,9 +160,10 @@ impl<T: ShaderParam> Batch for OwnedBatch<T> {
 type Index = u16;
 
 //#[derive(PartialEq, Eq, PartialOrd, Ord, Debug)]
+#[derive(Clone)]
 struct Id<T>(Index, PhantomData<T>);
 
-impl<T> Copy for Id<T> {}
+impl<T: Clone> Copy for Id<T> {}
```

Comment:
Could you instead do:
~~~rust
impl<T> Clone for Id<T> {
    fn clone(&self) -> Id<T> { *self }
}
~~~